tests/rados/mute_alerts.py

```diff
@@ -0,0 +1,238 @@
+import logging
+import re
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+
+log = logging.getLogger(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Polarion ID: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83573854
+    Feature : Mute Health alerts : https://trello.com/c/AU8FT6Qp/27-mute-health-warnings
+    RFE Bug : https://bugzilla.redhat.com/show_bug.cgi?id=1821508
+    upstream doc : https://docs.ceph.com/en/octopus/rados/operations/monitoring/#muting-health-checks
+
+    1. Check the cluster health status and simulate various failures.
+    2. Verify the health warning/ error generated due to the failures.
+    3. Mute the failures, test the various arguments and verify that the alerts are muted successfully
+    Args:
+        ceph_cluster (ceph.ceph.Ceph): ceph cluster
+    """
+    log.info("Running test for feature : Mute Health Alerts")
+    log.info(run.__doc__)
+    config = kw.get("config")
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    log.debug("Getting the health status of the cluster")
+    all_alerts = get_alerts(cephadm)
+    alert_list = ["MON_DISK_BIG", "OSDMAP_FLAGS"]
+    if not len(all_alerts["active_alerts"]) == 0:
+        log.info(
+            f"There are health alerts generated on the cluster. Alerts : {all_alerts['active_alerts']}"
+        )
+    else:
+        log.info("Cluster Health is ok. \n Generating a alert to verify feature")
+    alert = alert_list[0]
+    log.info(f"generating alert : {alert}")
+    if generate_health_alert(alert=alert, node=cephadm):
+        log.error("could not generate the error")
+        return 1
+    log.info(f"Proceeding to mute the alert: {alert}")
+    if mute_health_alert(alert=alert, node=cephadm, duration="6m"):
+        log.error(f"could not mute the alert : {alert}")
+        return 1
+    log.info(
+        "The alert was muted for 6 minutes. Checking if the alert is un-muted after the TTL"
+    )
+    time.sleep(362)
+    all_alerts = get_alerts(cephadm)
+    if alert not in all_alerts["active_alerts"] and alert in all_alerts["muted_alerts"]:
```

Comment:
The below steps are repeated thrice

- generated_health_alert
- mute_health_alert
- generate_health_alert

with varying variables or configurations. Would it be a good idea to move the common workflow to a separate function? We could call that method thrice with varying arguments.

Finally, we can have a single method to clear all the settings done... in this way we can ensure the settings are cleared even on failure